channels/tests/test_worker.py

```diff
@@ -35,3 +55,49 @@ def test_channel_filters(self):
             worker.apply_channel_filters(["yes.1", "no.1", "maybe.2", "yes"]),
             ["yes.1"],
         )
+
+    def test_run_with_consume_later_error(self):
+
+        # consumer with ConsumeLater error at first call
+        def _consumer(message, **kwargs):
+            if not hasattr(_consumer, '_called'):
+                _consumer._called = True
+                raise ConsumeLater()
+
+        consumer = mock.Mock(side_effect=_consumer)
```

Comment:
You seem to be using both mock.callcount and _consumer._called; is there a need for both? I'd recommend just combining call count into this inner part of the function and then you can drop the need to use mock and pass the consumer function in directly.